lib/kubernetes-deploy/sync_mediator.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class SyncMediator
+    def initialize(namespace:, context:, logger:)
+      @namespace = namespace
+      @context = context
+      @logger = logger
+      clear_cache
+    end
+
+    def get_instance(kind, resource_name)
+      if @cache.key?(kind)
+        @cache[kind].find { |r| r.dig("metadata", "name") == resource_name } || {}
```

Comment:
What do you think about the cache being a 2d array by kind and then name? (e.g. `@cache[kind][resource_name] || {}`)

Reply:
I think it's a good idea; preprocessing it into that form up front should in theory be more efficient overall.